internal/controller/resilient_deployment_workflow.go

```diff
@@ -0,0 +1,564 @@
+			// TODO: review the limitador event watch after the feature comes out of experimental
+			{Kind: &kuadrantv1beta1.LimitadorGroupKind},
+		},
+	}
+}
+
+func (r *ResilienceDeploymentPrecondition) run(ctx context.Context, _ []controller.ResourceEvent, topology *machinery.Topology, _ error, state *sync.Map) error {
+	logger := controller.LoggerFromContext(ctx).WithName("ResilienceDeploymentPrecondition")
+	logger.V(level).Info("ResilienceDeployment Precondition", "status", "started")
+	defer logger.V(level).Info("ResilienceDeployment Precondition", "status", "completed")
+
+	state.Store(ExperimentalResilienceFeature, isExperimentalFeatureEnabled(topology))
+
+	return nil
+}
+
+// INFO: Task Section
+
+func NewResilienceDeploymentTasks(client *dynamic.DynamicClient) []controller.ReconcileFunc {
+	return []controller.ReconcileFunc{
+		NewResilienceAuthorizationReconciler().Subscription().Reconcile,
+		NewResilienceCounterStorageReconciler(client).Subscription().Reconcile,
+		NewResilienceRateLimitingReconciler(client).Subscription().Reconcile,
+	}
+}
+
+func NewResilienceAuthorizationReconciler() *ResilienceAuthorizationReconciler {
+	return &ResilienceAuthorizationReconciler{}
+}
+
+type ResilienceAuthorizationReconciler struct{}
+
+func (r *ResilienceAuthorizationReconciler) Subscription() controller.Subscription {
+	return controller.Subscription{
+		ReconcileFunc: r.reconcile,
+		Events: []controller.ResourceEventMatcher{
+			{Kind: &kuadrantv1beta1.KuadrantGroupKind},
+		},
+	}
+}
+
+func (r *ResilienceAuthorizationReconciler) reconcile(ctx context.Context, _ []controller.ResourceEvent, _ *machinery.Topology, _ error, state *sync.Map) error {
+	logger := controller.LoggerFromContext(ctx).WithName("ResilienceAuthorizationReconciler")
+
+	logger.V(level).Info("ResilienceAuthorizationReconciler Task", "status", "started")
+	defer logger.V(level).Info("ResilienceAuthorizationReconciler Task", "status", "completed")
+	if !experimentalFeatureEnabledSate(state) {
+		logger.V(level).Info("Experimental resilience feature is not enabled, early exit", "status", "exiting")
+		return nil
+	}
+	logger.V(level).Info("Experimental resilience feature is enabled", "status", "processing")
+
+	return nil
+}
+
+func NewResilienceRateLimitingReconciler(client *dynamic.DynamicClient) *ResilienceRateLimitingReconciler {
+	return &ResilienceRateLimitingReconciler{Client: client}
+}
+
+type ResilienceRateLimitingReconciler struct {
+	Client *dynamic.DynamicClient
+}
+
+func (r *ResilienceRateLimitingReconciler) Subscription() controller.Subscription {
+	return controller.Subscription{
+		ReconcileFunc: r.reconcile,
+		Events: []controller.ResourceEventMatcher{
+			{Kind: &kuadrantv1beta1.KuadrantGroupKind},
+			{Kind: &kuadrantv1beta1.LimitadorGroupKind},
+		},
+	}
+}
+
+func (r *ResilienceRateLimitingReconciler) reconcile(ctx context.Context, _ []controller.ResourceEvent, topology *machinery.Topology, _ error, state *sync.Map) error {
```

Comment:
it would be nice to keep this method smaller. Too much done within it. This is not a request, just a "good to have" advice.

Reply:
I hear you. I guess this would be in the "Make it work" stage of "Make it work, Make it right, Make it right".